### PR TITLE
Feature/compression config

### PR DESCRIPTION
**Pull request by neolynx**
Make the compression method for deployments and inside the initrd installer configurable:
DEPLOYMENT_COMPRESSION = [ none, gzip, xz ]
INSTALLER_COMPRESSION = [ none, gzip, xz ]

### CODE REVIEW

**Review by hus0 — APPROVED** [review]

**Comment by hus0** [comment]
We should maybe use same variable for all deployments.

File: plugins/lxd.plugin [hunk 1]
```diff
@@ -48,7 +48,18 @@ EOF
 
   mv $WORK_DIR/root $WORK_DIR/rootfs
 
-  tar $TAR_PXZ -cf $WORK_DIR/${deployment} -C $WORK_DIR metadata.yaml templates rootfs
+  case $DEPLOYMENT_COMPRESSION in
+      none)
+         tar -cf $WORK_DIR/$deployment -C $WORK_DIR metadata.yaml templates rootfs
+      ;;
+      xz)
+         tar $TAR_PXZ -cJf $WORK_DIR/$deployment -C $WORK_DIR metadata.yaml templates rootfs
```
Comment (hus0) [inline plugins/lxd.plugin:56]: This doesn't work on stretch as TAR_PXZ is set to `pxz` but `J` option is used for xz.

**Review by hus0 — CHANGES_REQUESTED** [review]

File: molior-deploy [hunk 2]
```diff
@@ -681,6 +681,24 @@ setup_deployment()
...
+          DEPLOYMENT_COMPRESSION_EXT=""
+      ;;
+      xz)
+          DEPLOYMENT_COMPRESSION_EXT=".xz"
+      ;;
+      gzip)
+          DEPLOYMENT_COMPRESSION_EXT=".gz"
+      ;;
+      *)
+          log_error "Unsupported deployment compressions: '$DEPLOYMENT_COMPRESSION'"
```
Comment (randombenj) [inline molior-deploy:698]: Maybe add a comment to 'pleas use one of xz, gzip or none'

**Review by randombenj — APPROVED** [review]
Great!

**Pull request closed without merging**